httpx/_client.py

```diff
@@ -543,10 +543,13 @@ def _transport_for_url(self, url: URL) -> httpcore.SyncHTTPTransport:
         enforce_http_url(url)
 
         if self._proxies and not should_not_be_proxied(url):
-            is_default_port = (url.scheme == "http" and url.port == 80) or (
-                url.scheme == "https" and url.port == 443
+            default_port = {"http": 80, "https": 443}[url.scheme]
```

Comment:
Should this be a `.get()` to avoid `KeyError: 'ftp'`?
```suggestion
            default_port = {"http": 80, "https": 443}.get(url.scheme)
```

Reply:
Actually no, because we've called `enforce_http_url(url)` before doing anything else, so `url.scheme` will always be `http|https` here, and `default_port` will be an `int` rather than an `Optional[int]`.

(Agree that it's a bit futzy, and I think it'd potentially clean up a bit if we rethought the proxy_keys/mount API.)